src/app/views/loan-detail/loan-detail.component.ts

```diff
@@ -44,6 +53,25 @@ export class LoanDetailComponent implements OnInit {
     private spinner: NgxSpinnerService,
   ) {}
 
+  // Loan Oracle
+  private loanOracle(){
```

Comment:
The oracle and currency data can be read directly from loan.oracle and `loan.currency`, there is no loading required.